diff --git a/plugins/ultimate_cron/logger/database.class.php b/plugins/ultimate_cron/logger/database.class.php
index f7b5d46..fd97381 100644
--- a/plugins/ultimate_cron/logger/database.class.php
+++ b/plugins/ultimate_cron/logger/database.class.php
@@ -375,7 +375,15 @@ class UltimateCronDatabaseLogEntry extends UltimateCronLogEntry {
 
     static $retry = 0;
 
-    try {
+    $count = (int) db_select('ultimate_cron_log')
+      ->fields(NULL, array('field'))
+      ->condition('lid', $this->lid)
+      ->condition('end_time', 0)
+      ->countQuery()
+      ->execute()
+      ->fetchField();
+
+    if ($count == 0) {
       db_insert('ultimate_cron_log')
         ->fields(array(
           'lid' => $this->lid,
@@ -390,7 +398,7 @@ class UltimateCronDatabaseLogEntry extends UltimateCronLogEntry {
         ))
         ->execute();
     }
-    catch (PDOException $e) {
+    else {
       // Row already exists. Let's update it, if we can.
       $updated = db_update('ultimate_cron_log')
         ->fields(array(
@@ -418,7 +426,7 @@ class UltimateCronDatabaseLogEntry extends UltimateCronLogEntry {
         $retry++;
         if ($retry > 3) {
           $retry = 0;
-          watchdog('database_logger', (string) $e, array(), WATCHDOG_CRITICAL);
+          watchdog('database_logger', 'Could not write log entry', array(), WATCHDOG_CRITICAL);
           return;
         }
 
